enhancements/cloud-integration/openstack/failure-domains.md

```diff
@@ -0,0 +1,720 @@
+        direction TB
+   
+        control_plane_net_2(Control plane subnet 2)
+        storage_net_2(Storage subnet 2)
+ 
+        subgraph Nova AZ 2
+            control_2(Control plane node)
+        end
+ 
+        subgraph Cinder AZ 2
+            storage_2(Cinder storage)
+        end
+ 
+        control_2 -- tag: control_plane --- control_plane_net_2
+        control_2 -- tag: storage --- storage_net_2
+        storage_2 --- storage_net_2
+    end
+ 
+    subgraph Failure Domain 3
+        direction TB
+   
+        control_plane_net_3(Control plane subnet 3)
+        storage_net_3(Storage subnet 3)
+ 
+        subgraph Nova AZ 3
+            control_3(Control plane node)
+        end
+ 
+        subgraph Cinder AZ 3
+            storage_3(Cinder storage)
+        end
+ 
+        control_3 -- tag: control_plane --- control_plane_net_3
+        control_3 -- tag: storage --- storage_net_3
+        storage_3 --- storage_net_3
+    end
+ 
+    control_plane_net_1 --- spine
+    control_plane_net_2 --- spine
+    control_plane_net_3 --- spine
+```
+
+## Motivation
+
+This section is for explicitly listing the motivation, goals and non-goals of
+this proposal. Describe why the change is important and the benefits to users.
+
+### User Stories
+
+* As an administrator, I want to ensure my control plane is resilient to the
+  failure of a single compute availability zone.
+* As an administrator, I want to ensure my control plane is resilient to the
+  failure of a single storage availability zone.
+* As an administrator, I want to deploy my control plane across multiple
+  independent L2 networks.
+* As an administrator, when I implement failure domains for an OpenStack service
+  that did not previously have them, I want to be able to automatically redeploy
+  my control plane in a running cluster to take advantage of them.
+* As an administrator, when I modify a failure domain within my OpenStack
+  deployment, I want to be able to automatically redeploy my control plane to
+  take advantage of the change.
+* As an administrator who does not use failure domains, I do not want to
+  change any of my existing configuration or tooling due to this change.
+* As a developer of Control Plane MachineSet, I want to know unambiguously what
+  failure domain specification was used for a given Machine so I can ensure the
+  Machine is redeployed if required.
+* As a developer in the Shift on Stack team, I want to ensure that the OpenShift
+  definition of OpenStack failure domains is consistent with the
+  cluster-api-provider-openstack definition of OpenStack failure domains.
+
+### Goals
+
+* Define a unified failure domain specification for use by machines.
+* Integrate this into the installer.
+* Ensure this can be integrated into Control Plane MachineSet.
+
+### Non-Goals
+
+* Integration with Control Plane MachineSet.
+
+## Proposal
+
+In our proposed definition, a failure domain is a bundle of:
+* zero or one Compute zones
```

Comment:
Is there a reason why we want to have a max of 1 compute or storage zones is fd?  What will happen if you specify one or two of those three types of zones/port, and leave blank the others?